dot/node_test.go

```diff
@@ -0,0 +1,394 @@
+	err := nodeInstance.initNode(cfg)
+	require.NoError(t, err)
+
+	tests := []struct {
+		name     string
+		basepath string
+		want     bool
+	}{
+		{
+			name:     "blank base path",
+			basepath: "",
+			want:     false,
+		},
+		{
+			name:     "working example",
+			basepath: cfg.Global.BasePath,
+			want:     true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := NodeInitialized(tt.basepath)
+			assert.Equal(t, tt.want, got)
+		})
+	}
+}
+
+func initKeystore(t *testing.T, cfg *Config) (*keystore.GlobalKeystore, error) {
+	ks := keystore.NewGlobalKeystore()
+	// load built-in test keys if specified by `cfg.Account.Key`
+	err := keystore.LoadKeystore(cfg.Account.Key, ks.Acco)
+	require.NoError(t, err)
+
+	err = keystore.LoadKeystore(cfg.Account.Key, ks.Babe)
+	require.NoError(t, err)
+
+	err = keystore.LoadKeystore(cfg.Account.Key, ks.Gran)
+	require.NoError(t, err)
+
+	// if authority node, should have at least 1 key in keystore
+	if cfg.Core.Roles == types.AuthorityRole && (ks.Babe.Size() == 0 || ks.Gran.Size() == 0) {
+		return nil, ErrNoKeysProvided
+	}
+
+	return ks, nil
+}
+
+func TestNode_StartStop(t *testing.T) {
+	// TODO, skiping this test because it's hanging, fix this.
+	t.Skip()
+	serviceRegistryLogger := logger.New(log.AddContext("pkg", "services"))
+	type fields struct {
+		Name          string
+		Services      *services.ServiceRegistry
+		started       chan struct{}
+		metricsServer *metrics.Server
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		err    error
+	}{
+		{
+			name: "base case",
+			fields: fields{
+				Name:     "Node",
+				Services: services.NewServiceRegistry(serviceRegistryLogger),
+				started:  make(chan struct{}),
+			},
+			err: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			n := &Node{
+				Name:          tt.fields.Name,
+				Services:      tt.fields.Services,
+				started:       tt.fields.started,
+				metricsServer: tt.fields.metricsServer,
+			}
+
+			err := n.Start()
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+				<-n.started
+				n.Stop()
+			}
+		})
+	}
+}
+
+func Test_nodeBuilder_loadRuntime(t *testing.T) {
+	stateSrvc := newStateService(t)
+	cfg := NewTestConfig(t)
+
+	type args struct {
+		cfg       *Config
+		ns        *runtime.NodeStorage
+		stateSrvc *state.Service
+		ks        *keystore.GlobalKeystore
+		net       *network.Service
+	}
+	tests := []struct {
+		name string
+		args args
+		err  error
+	}{
+		{
+			name: "base case",
+			args: args{
+				cfg:       cfg,
+				ns:        &runtime.NodeStorage{},
+				stateSrvc: stateSrvc,
+				ks:        nil,
+				net:       nil,
+			},
+			err: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			no := nodeBuilder{}
+			err := no.loadRuntime(tt.args.cfg, tt.args.ns, tt.args.stateSrvc, tt.args.ks, tt.args.net)
+
+			assert.ErrorIs(t, err, tt.err)
```

Comment:
Should anything else be asserted? :thinking:

Reply:
Added check to confirm that code is stored in state storage.